db/memtable.cc

```diff
@@ -431,7 +434,16 @@ void MemTable::Add(SequenceNumber s, ValueType type,
   memcpy(p, value.data(), val_size);
   assert((unsigned)(p + val_size - buf) == (unsigned)encoded_len);
   if (!allow_concurrent) {
-    table->Insert(handle);
+    // Extract prefix for insert with hint.
+    Slice prefix;
+    if (insert_with_hint_prefix_extractor_ != nullptr) {
+      prefix = insert_with_hint_prefix_extractor_->Transform(key_slice);
+    }
+    if (prefix.empty()) {
```

Comment:
I want to provide a way for user to whitelist keys if they don't use hint for some of the keys (e.g. people mix keys with prefix and without prefix in the same db). Plus it doesn't make sense if prefix is empty.

In the updated revision I also check `prefix_extractor.InDomain()` and whitelist keys not in domain.